src/System.Net.Security/ref/System.Net.Security.cs

```diff
@@ -32,6 +32,8 @@ public enum EncryptionPolicy
         RequireEncryption = 0,
     }
     public delegate System.Security.Cryptography.X509Certificates.X509Certificate LocalCertificateSelectionCallback(object sender, string targetHost, System.Security.Cryptography.X509Certificates.X509CertificateCollection localCertificates, System.Security.Cryptography.X509Certificates.X509Certificate remoteCertificate, string[] acceptableIssuers);
+    public delegate System.Security.Cryptography.X509Certificates.X509Certificate ServerCertificateSelectionCallback(object sender, string hostName);
```

Comment:
Has this been API reviewed?  I'm wondering if we want to introduce a new delegate type or instead just use `Func<...>`.  Also, why `object` for sender?  What does someone do with that, and should it be more strongly typed?

Reply:
I would be more than happy with those changes. The only reason for object and a new delegate is convention nothing else.

Reply:
@stephentoub - The code has been API reviewed. `object` is not used in our code anywhere but it is a convention we chose in the past for all delegates - Func probably makes more sense but would stick out a little.